src/file.cpp

```diff
@@ -974,20 +968,28 @@ namespace libtorrent
 	{
 #if defined TORRENT_WINDOWS && !defined TORRENT_MINGW
 #if TORRENT_USE_WSTRING
-		wchar_t* cwd { ::_wgetcwd(nullptr, 0) };
-		std::shared_ptr<wchar_t> holder(cwd, &std::free);
-		std::string ret { cwd == nullptr ? "" : convert_from_wstring(cwd) };
+#define GetCurrentDir_ ::_wgetcwd
+#define PathCharType_ wchar_t
+#define ConvertFrom_ convert_from_wstring
 #else
-		char* cwd { ::_getcwd(nullptr, 0) };
-		std::shared_ptr<char> holder(cwd, &std::free);
-		std::string ret { cwd == nullptr ? "" : convert_from_native(cwd) };
+#define GetCurrentDir_ ::_getcwd
+#define PathCharType_ char
+#define ConvertFrom_ convert_from_native
 #endif // TORRENT_USE_WSTRING
 #else
-		char* cwd { ::getcwd(nullptr, 0) };
-		std::shared_ptr<char> holder(cwd, &std::free);
-		std::string ret { cwd == nullptr ? "/" : convert_from_native(cwd) };
-#endif
+#define GetCurrentDir_ ::getcwd
+#define PathCharType_ char
+#define ConvertFrom_ convert_from_native
+#endif
+		PathCharType_* cwd { GetCurrentDir_(nullptr, 0) };
```

Comment:
instead of specifying this type explicitly, you could say: ``auto cwd = GetCurrentDir_(nullptr, 0);``